sdk/communication/azure-communication-sms/src/main/java/com/azure/communication/sms/SmsClientBuilder.java

```diff
@@ -261,37 +254,58 @@ private HttpPipelinePolicy createHttpPipelineAuthPolicy() {
         }
     }
 
-    private HttpPipeline createHttpPipeline(HttpClient httpClient,
-                                            HttpPipelinePolicy authorizationPolicy,
-                                            HttpPipelinePolicy[] additionalPolicies) {
+    private HttpPipeline createHttpPipeline(HttpClient httpClient) {
+        if (this.pipeline != null) {
+            return this.pipeline;
+        }
+
+        List<HttpPipelinePolicy> policyList = new ArrayList<>();
+
+        ClientOptions buildClientOptions = (clientOptions == null) ? new ClientOptions() : clientOptions;
+        HttpLogOptions buildLogOptions = (httpLogOptions == null) ? new HttpLogOptions() : httpLogOptions;
+
+        String applicationId = null;
+        if (!CoreUtils.isNullOrEmpty(buildClientOptions.getApplicationId())) {
+            applicationId = buildClientOptions.getApplicationId();
+        } else if (!CoreUtils.isNullOrEmpty(buildLogOptions.getApplicationId())) {
+            applicationId = buildLogOptions.getApplicationId();
+        }
+
+        // Add required policies
+        policyList.add(this.createHttpPipelineAuthPolicy());
+        String clientName = properties.getOrDefault(SDK_NAME, "UnknownName");
+        String clientVersion = properties.getOrDefault(SDK_VERSION, "UnknownVersion");
+        policyList.add(new UserAgentPolicy(applicationId, clientName, clientVersion, configuration));
+        policyList.add(new RetryPolicy());
```

Comment:
Should not we create a new `RetryPolicy` only if `this.retryPolicy` is not set?

Reply:
Let's make sure other clients do the right thing as well

Reply:
True true, I will fix here and file bug.